dsafa/audio-band
# Add better unhandled exception detection

## Summary
Prevent the next startup when there is an unhandled exception. This should help mitigate crash loops that can occur due to toolbars being automatically started by windows.

## Checklist
- [x] Changes validated (manually or automated)
- [x] Closes / Fixes #167 (if applicable)

## Additional details / comments
This adds an application settings file with a boolean property `UnhandledException`.
In the `AppDomain.UnhandledException` event handler, set this property to `true`. And on every startup, check that property to see if there is an unhandled exception. If the property is true, then clear the value and prevent the startup.

If an exception is thrown at startup, then the toolbar will automatically be deactivated and won't start up automatically by explorer.

## Manual test steps (if applicable)
1. Force a crash that is not at startup.
2. Verify that audioband does not automatically start when explorer restarts.
